### ISSUE TITLE
Missing example response

### ISSUE THREAD

**Issue by stiancognite** [Mermade/widdershins#51]
Input `swagger.json`:
```json
{
  "swagger" : "2.0",
  "info" : {
    "description" : "This is documentation of my API",
    "version" : "0.3",
    "title" : "My API"
  },
  "host" : "my.url.com",
  "tags" : [ {
    "name" : "Somethings",
    "description" : "Endpoint for managing somethings"
  } ],
  "schemes" : [ "https" ],
  "paths" : {
    "/api/somethings" : {
      "get" : {
        "tags" : [ "Events" ],
        "summary" : "getSomethings",
        "description" : "Get list of somethings matching query",
        "operationId" : "getSomethings",
        "consumes" : [ "application/json" ],
        "produces" : [ "application/json" ],
        "parameters" : [ {
          "name" : "type",
          "in" : "query",
          "description" : "Type of something",
          "required" : false,
          "type" : "string"
        } ],
        "responses" : {
          "200" : {
            "description" : "successful operation",
            "schema" : {
              "$ref" : "#/definitions/EventResponse"
            }
          }
        },
        "security" : [ {
          "session" : [ ]
        }, {
          "api_key" : [ ]
        } ]
      }
    }
  },
  "securityDefinitions" : { },
  "definitions" : {
    "Event" : {
      "type" : "object",
      "properties" : {
        "id" : {
          "type" : "integer",
          "format" : "int64",
          "description" : "ID given by the API to the event."
        },
        "additionalData" : {
          "description" : "Customizable extra data about the event",
          "$ref" : "#/definitions/JsonElement"
        }
      },
      "description" : "An event represents something."
    },
    "EventResponse" : {
      "type" : "object",
      "properties" : {
        "data" : {
          "$ref" : "#/definitions/List_Event"
        }
      }
    },
    "JsonArray" : {
      "type" : "object",
      "properties" : {
        "asString" : {
          "type" : "string"
        },
        "asDouble" : {
          "type" : "number",
          "format" : "double"
        },
        "asInt" : {
          "type" : "integer",
          "format" : "int32"
        },
        "asBoolean" : {
          "type" : "boolean",
          "default" : false
        },
        "asNumber" : {
          "$ref" : "#/definitions/Number"
        },
        "asFloat" : {
          "type" : "number",
          "format" : "float"
        },
        "asLong" : {
          "type" : "integer",
          "format" : "int64"
        },
        "asByte" : {
          "type" : "string",
          "format" : "byte"
        },
        "asCharacter" : {
          "type" : "string"
        },
        "asBigDecimal" : {
          "type" : "number"
        },
        "asBigInteger" : {
          "type" : "integer"
        },
        "asShort" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonArray" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonPrimitive" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonNull" : {
          "type" : "boolean",
          "default" : false
        },
        "asJsonObject" : {
          "$ref" : "#/definitions/JsonObject"
        },
        "asJsonArray" : {
          "$ref" : "#/definitions/JsonArray"
        },
        "asJsonPrimitive" : {
          "$ref" : "#/definitions/JsonPrimitive"
        },
        "asJsonNull" : {
          "$ref" : "#/definitions/JsonNull"
        },
        "jsonObject" : {
          "type" : "boolean",
          "default" : false
        }
      }
    },
    "JsonElement" : {
      "type" : "object",
      "properties" : {
        "jsonArray" : {
          "type" : "boolean",
          "default" : false
        },
        "asString" : {
          "type" : "string"
        },
        "asDouble" : {
          "type" : "number",
          "format" : "double"
        },
        "asInt" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonPrimitive" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonNull" : {
          "type" : "boolean",
          "default" : false
        },
        "asJsonObject" : {
          "$ref" : "#/definitions/JsonObject"
        },
        "asJsonArray" : {
          "$ref" : "#/definitions/JsonArray"
        },
        "asJsonPrimitive" : {
          "$ref" : "#/definitions/JsonPrimitive"
        },
        "asJsonNull" : {
          "$ref" : "#/definitions/JsonNull"
        },
        "asBoolean" : {
          "type" : "boolean",
          "default" : false
        },
        "asNumber" : {
          "$ref" : "#/definitions/Number"
        },
        "asFloat" : {
          "type" : "number",
          "format" : "float"
        },
        "asLong" : {
          "type" : "integer",
          "format" : "int64"
        },
        "asByte" : {
          "type" : "string",
          "format" : "byte"
        },
        "asCharacter" : {
          "type" : "string"
        },
        "asBigDecimal" : {
          "type" : "number"
        },
        "asBigInteger" : {
          "type" : "integer"
        },
        "asShort" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonObject" : {
          "type" : "boolean",
          "default" : false
        }
      }
    },
    "JsonNull" : {
      "type" : "object",
      "properties" : {
        "jsonArray" : {
          "type" : "boolean",
          "default" : false
        },
        "asString" : {
          "type" : "string"
        },
        "asDouble" : {
          "type" : "number",
          "format" : "double"
        },
        "asInt" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonPrimitive" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonNull" : {
          "type" : "boolean",
          "default" : false
        },
        "asJsonObject" : {
          "$ref" : "#/definitions/JsonObject"
        },
        "asJsonArray" : {
          "$ref" : "#/definitions/JsonArray"
        },
        "asJsonPrimitive" : {
          "$ref" : "#/definitions/JsonPrimitive"
        },
        "asJsonNull" : {
          "$ref" : "#/definitions/JsonNull"
        },
        "asBoolean" : {
          "type" : "boolean",
          "default" : false
        },
        "asNumber" : {
          "$ref" : "#/definitions/Number"
        },
        "asFloat" : {
          "type" : "number",
          "format" : "float"
        },
        "asLong" : {
          "type" : "integer",
          "format" : "int64"
        },
        "asByte" : {
          "type" : "string",
          "format" : "byte"
        },
        "asCharacter" : {
          "type" : "string"
        },
        "asBigDecimal" : {
          "type" : "number"
        },
        "asBigInteger" : {
          "type" : "integer"
        },
        "asShort" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonObject" : {
          "type" : "boolean",
          "default" : false
        }
      }
    },
    "JsonObject" : {
      "type" : "object",
      "properties" : {
        "jsonArray" : {
          "type" : "boolean",
          "default" : false
        },
        "asString" : {
          "type" : "string"
        },
        "asDouble" : {
          "type" : "number",
          "format" : "double"
        },
        "asInt" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonPrimitive" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonNull" : {
          "type" : "boolean",
          "default" : false
        },
        "asJsonObject" : {
          "$ref" : "#/definitions/JsonObject"
        },
        "asJsonArray" : {
          "$ref" : "#/definitions/JsonArray"
        },
        "asJsonPrimitive" : {
          "$ref" : "#/definitions/JsonPrimitive"
        },
        "asJsonNull" : {
          "$ref" : "#/definitions/JsonNull"
        },
        "asBoolean" : {
          "type" : "boolean",
          "default" : false
        },
        "asNumber" : {
          "$ref" : "#/definitions/Number"
        },
        "asFloat" : {
          "type" : "number",
          "format" : "float"
        },
        "asLong" : {
          "type" : "integer",
          "format" : "int64"
        },
        "asByte" : {
          "type" : "string",
          "format" : "byte"
        },
        "asCharacter" : {
          "type" : "string"
        },
        "asBigDecimal" : {
          "type" : "number"
        },
        "asBigInteger" : {
          "type" : "integer"
        },
        "asShort" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonObject" : {
          "type" : "boolean",
          "default" : false
        }
      }
    },
    "JsonPrimitive" : {
      "type" : "object",
      "properties" : {
        "boolean" : {
          "type" : "boolean",
          "default" : false
        },
        "number" : {
          "type" : "boolean",
          "default" : false
        },
        "string" : {
          "type" : "boolean",
          "default" : false
        },
        "asString" : {
          "type" : "string"
        },
        "asDouble" : {
          "type" : "number",
          "format" : "double"
        },
        "asInt" : {
          "type" : "integer",
          "format" : "int32"
        },
        "asBoolean" : {
          "type" : "boolean",
          "default" : false
        },
        "asNumber" : {
          "$ref" : "#/definitions/Number"
        },
        "asFloat" : {
          "type" : "number",
          "format" : "float"
        },
        "asLong" : {
          "type" : "integer",
          "format" : "int64"
        },
        "asByte" : {
          "type" : "string",
          "format" : "byte"
        },
        "asCharacter" : {
          "type" : "string"
        },
        "asBigDecimal" : {
          "type" : "number"
        },
        "asBigInteger" : {
          "type" : "integer"
        },
        "asShort" : {
          "type" : "integer",
          "format" : "int32"
        },
        "jsonArray" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonPrimitive" : {
          "type" : "boolean",
          "default" : false
        },
        "jsonNull" : {
          "type" : "boolean",
          "default" : false
        },
        "asJsonObject" : {
          "$ref" : "#/definitions/JsonObject"
        },
        "asJsonArray" : {
          "$ref" : "#/definitions/JsonArray"
        },
        "asJsonPrimitive" : {
          "$ref" : "#/definitions/JsonPrimitive"
        },
        "asJsonNull" : {
          "$ref" : "#/definitions/JsonNull"
        },
        "jsonObject" : {
          "type" : "boolean",
          "default" : false
        }
      }
    },
    "List_Event" : {
      "type" : "object",
      "properties" : {
        "items" : {
          "type" : "array",
          "items" : {
            "$ref" : "#/definitions/Event"
          }
        }
      }
    }
  }
}
```

Output of `widdershins swagger.json -o index.html.md`:
```markdown
---
title: My API v0.3
language_tabs:
  - shell: Shell
  - http: HTTP
  - javascript: JavaScript
  - javascript--nodejs: Node.JS
  - ruby: Ruby
  - python: Python
  - java: Java
toc_footers: []
includes: []
search: true
highlight_theme: darkula
---

# My API v0.3

> Scroll down for code samples, example requests and responses. Select a language for code samples from the tabs above or the mobile navigation menu.

This is documentation of my API

Base URLs:

* <a href="https://my.url.com/">https://my.url.com/</a>





# Events

## getSomethings

> Code samples

/```shell
# You can also use wget
curl -X GET https://my.url.com//api/somethings \
  -H 'Accept: application/json'

/```

/```http
GET https://my.url.com//api/somethings HTTP/1.1
Host: my.url.com

Accept: application/json

/```

/```javascript
var headers = {
  'Accept':'application/json'

};

$.ajax({
  url: 'https://my.url.com//api/somethings',
  method: 'get',

  headers: headers,
  success: function(data) {
    console.log(JSON.stringify(data));
  }
})
/```

/```javascript--nodejs
const request = require('node-fetch');

const headers = {
  'Accept':'application/json'

};

fetch('https://my.url.com//api/somethings',
{
  method: 'GET',

  headers: headers
})
.then(function(res) {
    return res.json();
}).then(function(body) {
    console.log(body);
});
/```

/```ruby
require 'rest-client'
require 'json'

headers = {
  'Accept' => 'application/json'
}

result = RestClient.get 'https://my.url.com//api/somethings',
  params: {
  }, headers: headers

p JSON.parse(result)
/```

/```python
import requests
headers = {
  'Accept': 'application/json'
}

r = requests.get('https://my.url.com//api/somethings', params={

}, headers = headers)

print r.json()
/```

/```java
URL obj = new URL("https://my.url.com//api/somethings");
HttpURLConnection con = (HttpURLConnection) obj.openConnection();
con.setRequestMethod("GET");
int responseCode = con.getResponseCode();
BufferedReader in = new BufferedReader(
    new InputStreamReader(con.getInputStream()));
String inputLine;
StringBuffer response = new StringBuffer();
while ((inputLine = in.readLine()) != null) {
    response.append(inputLine);
}
in.close();
System.out.println(response.toString());
/```

`GET /api/somethings`

*getSomethings*

Get list of somethings matching query

### Parameters

Parameter|In|Type|Required|Description
---|---|---|---|---|
type|query|string|false|Type of something


> Example responses

/```json
{
  "data": {}
}
/```
### Responses

Status|Meaning|Description|Schema
---|---|---|---|
200|[OK](https://tools.ietf.org/html/rfc7231#section-6.3.1)|successful operation|[EventResponse](#schemaeventresponse)

<aside class="warning">
To perform this operation, you must be authenticated by means of one of the following methods:
None, None
</aside>

# Schemas

## Event

<a name="schemaevent"></a>

/```json
{
  "id": 0,
  "additionalData": {}
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
id|integer(int64)|false|ID given by the API to the event.
additionalData|[JsonElement](#schemajsonelement)|false|No description



## EventResponse

<a name="schemaeventresponse"></a>

/```json
{
  "data": {}
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
data|[List_Event](#schemalist_event)|false|No description



## JsonArray

<a name="schemajsonarray"></a>

/```json
{
  "asString": "string",
  "asDouble": 0,
  "asInt": 0,
  "asBoolean": false,
  "asNumber": {},
  "asFloat": 0,
  "asLong": 0,
  "asByte": "string",
  "asCharacter": "string",
  "asBigDecimal": 0,
  "asBigInteger": 0,
  "asShort": 0,
  "jsonArray": false,
  "jsonPrimitive": false,
  "jsonNull": false,
  "asJsonObject": {},
  "asJsonArray": {},
  "asJsonPrimitive": {},
  "asJsonNull": {},
  "jsonObject": false
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
asString|string|false|No description
asDouble|number(double)|false|No description
asInt|integer(int32)|false|No description
asBoolean|boolean|false|No description
asNumber|[](#schema)|false|No description
asFloat|number(float)|false|No description
asLong|integer(int64)|false|No description
asByte|string(byte)|false|No description
asCharacter|string|false|No description
asBigDecimal|number|false|No description
asBigInteger|integer|false|No description
asShort|integer(int32)|false|No description
jsonArray|boolean|false|No description
jsonPrimitive|boolean|false|No description
jsonNull|boolean|false|No description
asJsonObject|[JsonObject](#schemajsonobject)|false|No description
asJsonArray|[JsonArray](#schemajsonarray)|false|No description
asJsonPrimitive|[JsonPrimitive](#schemajsonprimitive)|false|No description
asJsonNull|[JsonNull](#schemajsonnull)|false|No description
jsonObject|boolean|false|No description



## JsonElement

<a name="schemajsonelement"></a>

/```json
{
  "jsonArray": false,
  "asString": "string",
  "asDouble": 0,
  "asInt": 0,
  "jsonPrimitive": false,
  "jsonNull": false,
  "asJsonObject": {},
  "asJsonArray": {},
  "asJsonPrimitive": {},
  "asJsonNull": {},
  "asBoolean": false,
  "asNumber": {},
  "asFloat": 0,
  "asLong": 0,
  "asByte": "string",
  "asCharacter": "string",
  "asBigDecimal": 0,
  "asBigInteger": 0,
  "asShort": 0,
  "jsonObject": false
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
jsonArray|boolean|false|No description
asString|string|false|No description
asDouble|number(double)|false|No description
asInt|integer(int32)|false|No description
jsonPrimitive|boolean|false|No description
jsonNull|boolean|false|No description
asJsonObject|[JsonObject](#schemajsonobject)|false|No description
asJsonArray|[JsonArray](#schemajsonarray)|false|No description
asJsonPrimitive|[JsonPrimitive](#schemajsonprimitive)|false|No description
asJsonNull|[JsonNull](#schemajsonnull)|false|No description
asBoolean|boolean|false|No description
asNumber|[](#schema)|false|No description
asFloat|number(float)|false|No description
asLong|integer(int64)|false|No description
asByte|string(byte)|false|No description
asCharacter|string|false|No description
asBigDecimal|number|false|No description
asBigInteger|integer|false|No description
asShort|integer(int32)|false|No description
jsonObject|boolean|false|No description



## JsonNull

<a name="schemajsonnull"></a>

/```json
{
  "jsonArray": false,
  "asString": "string",
  "asDouble": 0,
  "asInt": 0,
  "jsonPrimitive": false,
  "jsonNull": false,
  "asJsonObject": {},
  "asJsonArray": {},
  "asJsonPrimitive": {},
  "asJsonNull": {},
  "asBoolean": false,
  "asNumber": {},
  "asFloat": 0,
  "asLong": 0,
  "asByte": "string",
  "asCharacter": "string",
  "asBigDecimal": 0,
  "asBigInteger": 0,
  "asShort": 0,
  "jsonObject": false
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
jsonArray|boolean|false|No description
asString|string|false|No description
asDouble|number(double)|false|No description
asInt|integer(int32)|false|No description
jsonPrimitive|boolean|false|No description
jsonNull|boolean|false|No description
asJsonObject|[JsonObject](#schemajsonobject)|false|No description
asJsonArray|[JsonArray](#schemajsonarray)|false|No description
asJsonPrimitive|[JsonPrimitive](#schemajsonprimitive)|false|No description
asJsonNull|[JsonNull](#schemajsonnull)|false|No description
asBoolean|boolean|false|No description
asNumber|[](#schema)|false|No description
asFloat|number(float)|false|No description
asLong|integer(int64)|false|No description
asByte|string(byte)|false|No description
asCharacter|string|false|No description
asBigDecimal|number|false|No description
asBigInteger|integer|false|No description
asShort|integer(int32)|false|No description
jsonObject|boolean|false|No description



## JsonObject

<a name="schemajsonobject"></a>

/```json
{
  "jsonArray": false,
  "asString": "string",
  "asDouble": 0,
  "asInt": 0,
  "jsonPrimitive": false,
  "jsonNull": false,
  "asJsonObject": {},
  "asJsonArray": {},
  "asJsonPrimitive": {},
  "asJsonNull": {},
  "asBoolean": false,
  "asNumber": {},
  "asFloat": 0,
  "asLong": 0,
  "asByte": "string",
  "asCharacter": "string",
  "asBigDecimal": 0,
  "asBigInteger": 0,
  "asShort": 0,
  "jsonObject": false
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
jsonArray|boolean|false|No description
asString|string|false|No description
asDouble|number(double)|false|No description
asInt|integer(int32)|false|No description
jsonPrimitive|boolean|false|No description
jsonNull|boolean|false|No description
asJsonObject|[JsonObject](#schemajsonobject)|false|No description
asJsonArray|[JsonArray](#schemajsonarray)|false|No description
asJsonPrimitive|[JsonPrimitive](#schemajsonprimitive)|false|No description
asJsonNull|[JsonNull](#schemajsonnull)|false|No description
asBoolean|boolean|false|No description
asNumber|[](#schema)|false|No description
asFloat|number(float)|false|No description
asLong|integer(int64)|false|No description
asByte|string(byte)|false|No description
asCharacter|string|false|No description
asBigDecimal|number|false|No description
asBigInteger|integer|false|No description
asShort|integer(int32)|false|No description
jsonObject|boolean|false|No description



## JsonPrimitive

<a name="schemajsonprimitive"></a>

/```json
{
  "boolean": false,
  "number": false,
  "string": false,
  "asString": "string",
  "asDouble": 0,
  "asInt": 0,
  "asBoolean": false,
  "asNumber": {},
  "asFloat": 0,
  "asLong": 0,
  "asByte": "string",
  "asCharacter": "string",
  "asBigDecimal": 0,
  "asBigInteger": 0,
  "asShort": 0,
  "jsonArray": false,
  "jsonPrimitive": false,
  "jsonNull": false,
  "asJsonObject": {},
  "asJsonArray": {},
  "asJsonPrimitive": {},
  "asJsonNull": {},
  "jsonObject": false
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
boolean|boolean|false|No description
number|boolean|false|No description
string|boolean|false|No description
asString|string|false|No description
asDouble|number(double)|false|No description
asInt|integer(int32)|false|No description
asBoolean|boolean|false|No description
asNumber|[](#schema)|false|No description
asFloat|number(float)|false|No description
asLong|integer(int64)|false|No description
asByte|string(byte)|false|No description
asCharacter|string|false|No description
asBigDecimal|number|false|No description
asBigInteger|integer|false|No description
asShort|integer(int32)|false|No description
jsonArray|boolean|false|No description
jsonPrimitive|boolean|false|No description
jsonNull|boolean|false|No description
asJsonObject|[JsonObject](#schemajsonobject)|false|No description
asJsonArray|[JsonArray](#schemajsonarray)|false|No description
asJsonPrimitive|[JsonPrimitive](#schemajsonprimitive)|false|No description
asJsonNull|[JsonNull](#schemajsonnull)|false|No description
jsonObject|boolean|false|No description



## List_Event

<a name="schemalist_event"></a>

/```json
{
  "items": [
    {}
  ]
} 
/```

### Properties

Name|Type|Required|Description
---|---|---|---|
type|Unknown|false|No description
items|[Event](#schemaevent)|false|No description
x-isArray|Unknown|false|No description
$ref|Unknown|false|No description
```

What I expect to see is a response body, not a `data` field with an empty object. Is there are problem when widdershins is parsing `JsonElement` that causes everything else in my `Event` class to disappear? Because even though I add other fields they don't appear unless I remove the `JsonElement` field.

**Comment by stiancognite** [comment]
Some background: I'm using various types from `com.google.gson` in my Java application. Any use of those types in one of my API types seems to cause this problem when using widdershins. I'm using widdershins version 2.3.

**Comment by MikeRalphson** [comment]
When you say "what I expect to see is a response body", which type from `#/definitions/` are you expecting to see, and do you mean as an example of the 200 response?

I think what we have here is a combination of circular definitions (which Widdershins doesn't handle well yet), objects with no required properties (so openapi-sampler is returning empty objects as valid examples) and another bug (the handling of the `$ref` in array.items of List_Event).

**Comment by stiancognite** [comment]
I expect to see a list of `List_Event`'s, i.e. a list of `Event`'s (with its properties id (integer), and additionalData (JsonElement).

Thank you for responding so fast!

**Comment by MikeRalphson** [comment]
I've published v2.2.3 with a new option `-a` or `--aggressive` which enables a more aggressive dereferencing of circular refs. It isn't the default as it can cause infinite loops on some schemas (I'm still looking if there are ways to detect this).

Also fixed are the erroneous `type`, `x-isArray` and `$ref` seen at the end of your example:

```markdown
Name|Type|Required|Description
---|---|---|---|
type|Unknown|false|No description
items|[Event](#schemaevent)|false|No description
x-isArray|Unknown|false|No description
$ref|Unknown|false|No description
```

Let me know if this improves things for you...

**Comment by stiancognite** [comment]
It seems to get stuck. I tried `widdershins swagger.json -o index.html.md -a`, `widdershins swagger.json -a -o index.html.md`, and `widdershins -a swagger.json -o index.html.md` just to make sure. I get this output:
```
➜  swagger git:(master) widdershins -a swagger.json -o index.html.md                                                                                                            │ * Documentation:  https://help.ubuntu.com
Compiling all doT templates...                                                                                                                                                  │ * Management:     https://landscape.canonical.com
Compiling authentication.dot to function                                                                                                                                        │ * Support:        https://ubuntu.com/advantage
Compiling authentication_none.dot to function                                                                                                                                   │Last login: Thu Oct  5 12:02:59 2017 from 10.108.0.1
Compiling code_http.dot to function                                                                                                                                             │cognite@bastion-host-1775741931-70t03:~$ packet_write_wait: Connection to 35.189.212.92 port 22: Broken pi
Compiling code_java.dot to function                                                                                                                                             │pe
Compiling code_javascript.dot to function                                                                                                                                       │➜  ~ sshc
Compiling code_nodejs.dot to function                                                                                                                                           │Welcome to Ubuntu 16.04.2 LTS (GNU/Linux 4.4.52+ x86_64)
Compiling code_python.dot to function                                                                                                                                           │
Compiling code_ruby.dot to function                                                                                                                                             │ * Documentation:  https://help.ubuntu.com
Compiling code_shell.dot to function                                                                                                                                            │ * Management:     https://landscape.canonical.com
Compiling discovery.dot to function                                                                                                                                             │ * Support:        https://ubuntu.com/advantage
Compiling footer.dot to function                                                                                                                                                │Last login: Thu Oct  5 20:56:47 2017 from 10.108.0.1
Compiling heading_body_parameter.dot to function                                                                                                                                │cognite@bastion-host-1775741931-70t03:~$ packet_write_wait: Connection to 35.189.212.92 port 22: Broken pi
Compiling heading_code_samples.dot to function                                                                                                                                  │pe
Compiling heading_example_responses.dot to function                                                                                                                             │➜  ~
Compiling heading_main.dot to function                                                                                                                                          │
Compiling parameters.dot to function                                                                                                                                            │
Compiling response_headers.dot to function                                                                                                                                      │
Compiling response_schema.dot to function                                                                                                                                       │
Compiling responses.dot to function                                                                                                                                             │
Compiling schema_header.dot to function                                                                                                                                         │
Compiling schema_properties.dot to function                                                                                                                                     │
Compiling schema_sample.dot to function                                                                                                                                         │
Compiling security.dot to function
```
but it never completes. So that's probably an infinite loop in my schema?

**Comment by MikeRalphson** [comment]
> So that's probably an infinite loop in my schema?

Looks like it... Though your example above completes with `--aggressive` here, I understand your real definition is probably more complex. Let me have another think...

**Comment by stiancognite** [comment]
I think we might use something that shows docs directly from `swagger.json` instead of using widdershins->slate, so please don't use more time on this if it's only for my issue. :) Thank you for the help!

**Comment by MikeRalphson** [comment]
Understood. Please let me know which tools you consider / end up going with as it would be very useful to see their approaches.

**Comment by stiancognite** [comment]
[swagger-ui](https://github.com/swagger-api/swagger-ui) or [ReDoc](https://github.com/Rebilly/ReDoc) seems the most likely candidates. I can report back when we decide.

**Comment by MikeRalphson** [comment]
ReDoc is excellent, I can heartily recommend it. I hope it works for you.

**Comment by stianlagstad** [comment]
I've tried a bunch of alternatives:
- ReDoc: Looks good. Handles GSON types well. Is slow to parse and present our schema. Is slow/buggy to use when done parsing (our schema might be too big?). Might work well enough for us when [their rewrite to React](https://github.com/Rebilly/ReDoc/issues/327) is done.
- Swagger-UI: Crashes when loading our schema (our schema isn't _that_ big). Unable to use.
- [bootprint](https://github.com/bootprint/bootprint-openapi): Works well (because it doesn't rely on loading and parsing swagger.json with javascript?). But it doesn't look so good.
- [swagger-to-slate](https://github.com/lavkumarv/swagger-to-slate): Unsatisfactory as it doesn't show response examples. (There's probably more stuff missing from this one as well..)
- [spectacle](https://github.com/sourcey/spectacle): Seems like a good alternative! No problems showing GSON types. Fast. Looks sliiiightly worse than slate.

Not sure what we're going to do yet. We might try to avoid using those GSON types that gives us problems with widdershins.

**Comment by MikeRalphson** [comment]
Thanks for taking the time to let me know your findings.

**Comment by MikeRalphson** [comment]
Closing this now as v3.0.0 includes completely different dereferencing and schema extraction algorithms.

**Comment by fraggle222** [comment]
Why is this closed. Example responses are not there at all. Shows fine if using swagger.io editor.

So in this swagger json for example, the response 200 clearly has a schema. And that definition is also in the json (and has no other references, no circular references).  
`	"responses": {
					"200": {
						"description": "OK",
						"schema": {
							"$ref": "#/definitions/OpenResponse"
						}
					},
					"201": {
						"description": "Created"
					},
					"401": {
						"description": "Unauthorized"
					},
					"403": {
						"description": "Forbidden"
					},
					"404": {
						"description": "Not Found"
					}
				}`

After widdershins conversion, that $ref": "#/definitions/OpenResponse" is never shown in the example responses. Widdershins generates no output for it.  It does show as a table in the main section for "Responses", but no "Example Responses" output.

Bottom line Widdershins fails to generate example responses markdown.

**Comment by MikeRalphson** [comment]
Thanks for the further info!

**Comment by MikeRalphson** [comment]
Sorry, perhaps we both had suboptimal days? Apologies.

**Comment by Eileen0917** [comment]
Hi @MikeRalphson,

Is it possible to custom the response example?
I want it to be like what it show on swagger ( like the picture shown below )
![right example](https://imgur.com/dbfpHjk.jpg)

what I get now:
![not expected example](https://imgur.com/W6c6zEQ.jpg)

Thanks!
